React/Views/RCTSliderManager.m

```diff
@@ -30,6 +30,13 @@ - (UIView *)view
 
 static void RCTSendSliderEvent(RCTSlider *sender, BOOL continuous)
 {
+  if (sender.step > 0 &&
+    sender.step <= (sender.maximumValue - sender.minimumValue)) {
+    int step = ((int)((sender.value + sender.step / 2) / sender.step) * sender.step);
```

Comment:
The step value is a float, so I'm not sure you should assume that it's a whole number. This logic should probably be:

    float value = round(sender.value / sender.step) * sender.step;